sunpy/data/_sample.py

```diff
@@ -166,3 +173,31 @@ def download_sample_data(overwrite=False):
             _handle_final_errors(results)
 
     return results + already_downloaded
+
+
+def _get_sample_files(filename_list, no_download=False, force_download=False):
+    sampledata_dir = _get_sampledata_dir()
+
+    fullpath = [sampledata_dir/fn for fn in filename_list]
+
+    if no_download:
+        fullpath = [fp if fp.exists() else None for fp in fullpath]
+    else:
+        to_download = zip(filename_list, fullpath)
+        if not force_download:
+            to_download = [(fn, fp) for fn, fp in to_download if not fp.exists()]
+
+        if to_download:
```

Comment:
```suggestion
        if len(to_download):
```
To make it clear that `to_download` is an iterable and not a `bool`.

Reply:
I understand your perspective, but I'll point out that the PEP 8 recommendation – which I don't always agree with – is to not have the additional `len()` when checking for empty sequences.

Reply:
Is there a reason to check the length of it instead of just doing a plain if?

Reply:
I think @dstansby isn't particularly comfortable with the falsiness of empty sequences, which is a fair position

Reply:
Is it bad practice? I use it all the time in my code.

Reply:
As I said, PEP 8 recommends not using `len()`, so it's certainly not "bad practice"

Reply:
👍 happy to keep this and follow PEP8